From 259a3d8257fc439d74a0a31036c4f5796e296a44 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@cryptomilk.org>
Date: Mon, 22 Jan 2018 07:47:08 +0100
Subject: [PATCH 09/12] storage: Allow to set sdcards to visible

If they are visibile to apps the media scanner will scan them.

Change-Id: Icceceafc8469b6b9d1c0e72f8425d7e38ae4a10b
---
 core/java/android/os/storage/StorageManager.java                 | 2 ++
 services/core/java/com/android/server/StorageManagerService.java | 7 +++++++
 2 files changed, 9 insertions(+)

diff --git a/core/java/android/os/storage/StorageManager.java b/core/java/android/os/storage/StorageManager.java
index 8533c7e..46de286 100644
--- a/core/java/android/os/storage/StorageManager.java
+++ b/core/java/android/os/storage/StorageManager.java
@@ -120,6 +120,8 @@ public class StorageManager {
     public static final String PROP_VIRTUAL_DISK = "persist.sys.virtual_disk";
     /** {@hide} */
     public static final String PROP_ADOPTABLE_FBE = "persist.sys.adoptable_fbe";
+    /** {@hide} */
+    public static final String PROP_SDCARD_VISIBLE = "persist.sys.sdcard_visible";
 
     /** {@hide} */
     public static final String UUID_PRIVATE_INTERNAL = null;
diff --git a/services/core/java/com/android/server/StorageManagerService.java b/services/core/java/com/android/server/StorageManagerService.java
index 1ca0112..8ee85d5 100644
--- a/services/core/java/com/android/server/StorageManagerService.java
+++ b/services/core/java/com/android/server/StorageManagerService.java
@@ -1320,6 +1320,13 @@ class StorageManagerService extends IStorageManager.Stub
                 vol.mountFlags |= VolumeInfo.MOUNT_FLAG_VISIBLE;
             }
 
+            // Set sdcards to visible to apps. If they are visible media is scanned
+            // and they can be used for other stuff.
+            if (SystemProperties.getBoolean(StorageManager.PROP_SDCARD_VISIBLE, false) &&
+                vol.disk.isSd()) {
+                vol.mountFlags |= VolumeInfo.MOUNT_FLAG_VISIBLE;
+            }
+
             vol.mountUserId = mCurrentUserId;
             mHandler.obtainMessage(H_VOLUME_MOUNT, vol).sendToTarget();
 
-- 
2.7.4

